### PR TITLE
ML - Fixed a bug with validating empty table name

### LINKED ISSUE #13257: The Table meets requirement check in ML import model doesnt behave correctly
- Azure Data Studio Version:

Steps to Reproduce:

1. Issue 1 : Go to the import model table selection page. the box is there visible without any check mark which looks confiusing 
2. Issue 2 : Select new table, type name then remove name : the green check is still visible and import button an available - both shouldnt be. If I click import it fails with red ribbon as expected.

Images below
Image 1
image

Image 2
image

### PR DESCRIPTION
**Pull request by llali**
This PR fixes #13257

### CODE REVIEW

File: extensions/machine-learning/src/views/models/tableSelectionComponent.ts [hunk 1]
```diff
@@ -209,7 +213,7 @@ export class TableSelectionComponent extends ModelViewBase implements IDataCompo
 	}
 
 	public get isDataValid(): boolean {
-		return this.data !== undefined && this.data.databaseName !== this._settings.defaultDbName && this.data.tableName !== this._settings.defaultTableName;
+		return this.data !== undefined && this.data.databaseName !== '' && this.data.tableName !== '' && this.data.databaseName !== undefined && this.data.tableName !== undefined && this.data.databaseName !== this._settings.defaultDbName && this.data.tableName !== this._settings.defaultTableName;
```
Comment (udeeshagautam) [inline extensions/machine-learning/src/views/models/tableSelectionComponent.ts:216]: can we also make sure that the empty square box isnt visible when user just lands on the page (Issue 1 in the bug)

Comment (llali) [reply extensions/machine-learning/src/views/models/tableSelectionComponent.ts:216]: that's similar to current design. 
Screen Shot 2020-11-05 at 2 52 27 PM
maybe we should revisit?

Comment (udeeshagautam) [reply extensions/machine-learning/src/views/models/tableSelectionComponent.ts:216]: this box looks bit more grey and has 'leran more' but the actual box doesnt but I agree with you that lets not remove it for this release if this is as per UX mockups and we should revisit post this release. This box totally confused me - doesnt seem like we need it atleast till we have 'learn more'. lets discuss next week.

@MsSQLGirl  FYI

**Review by udeeshagautam — APPROVED** [review]
Signing off for the fix. We will discuss the UX ater.

**Pull request merged**